src/Sarif/NotYetAutoGenerated/Result.cs

```diff
@@ -83,26 +92,41 @@ public ResultKind Kind
             }
         }
 
-        private FailureLevel _level;
+        private FailureLevel? _level;
 
         /// <summary>
         /// A value specifying the severity level of the result.
         /// </summary>
         [DataMember(Name = "level", IsRequired = false, EmitDefaultValue = false)]
-        [DefaultValue(FailureLevel.Warning)]
-        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
         [JsonConverter(typeof(Microsoft.CodeAnalysis.Sarif.Readers.EnumConverter))]
-        public FailureLevel Level
+        public FailureLevel? Level
         {
-            get { return _level; }
-            set
+            get
             {
-                _level = value;
-                if (_level != FailureLevel.None)
+                if (_level.HasValue)
                 {
-                    _kind = ResultKind.Fail;
+                    return _level;
                 }
```

Comment:
this is moved to the get() of Kind, and it need to consider default configuration.

related test:
ResultTests.cs: 
Result_Level_ResetsKindValue_WithDefaultConfiguration()